jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -638,17 +653,30 @@ public Request.Processor handle(Request request) throws Exception
         // wrap might fail (eg ServletContextHandler could not match a servlet)
         if (contextRequest == null)
             return null;
+        request.setAttribute(ContextRequest.class.getName(), contextRequest);
 
         // Does this handler want to process the request itself?
         Request.Processor processor = processByContextHandler(contextRequest);
         if (processor != null)
             return processor;
 
-        // The contextRequest is-a Supplier<Processor> that calls effectively calls getHandler().handle(request).
-        // Call this supplier in the scope of the context.
-        Request.Processor contextScopedProcessor = _context.get(contextRequest, contextRequest);
-        // Wrap the contextScopedProcessor with a wrapper that uses the wrapped request
-        return contextRequest.wrapProcessor(contextScopedProcessor);
```

Comment:
This is a prime example of confusing code that this PR replaces.   It is entirely unclear what line 649 is doing, specifically that it is ultimately doing a `processor = getHandler().handle(contextRequest)` and that the `Processor ContextRequest#get()` method should be looked at for the flow of control.

It is further unclear that line 651 returns the contextRequest as the wrapped processor, but with the normal process method extended, so that the request is ultimately used as a `Runnable` passed to `context.run(Runnable)`

This is very convoluted code, just to avoid some allocation of cheap wrappers.